app/models/patient.rb

```diff
@@ -133,6 +133,39 @@ def to_bundle(base_url)
     bundle
   end
 
+  # compares model's fhir attributes to the key-value pairs in hash
+  def match?(hash)
+    matching = true
+    hash.each do |k, v|
+      if has_attribute? k
+        matching &&= (send(k).to_s == v.to_s)
+        break unless matching
```

Comment:
Rather than using `each` with `break`:
```
hash
  .select { |key, _| has_attribute? key }
  .all? { |key, value| send(key).to_s == value.to_s }
```
Is this really robust for complex types?

Reply:
I think using `===` is better for complex types but then rubocop fails the code, however using `==` directly fails in functionality. Not sure if there are any other suggestions.

Reply:
I just did some testing and it seems like fhir models compares complex types at least somewhat intelligently, so we should be fine with `==`.